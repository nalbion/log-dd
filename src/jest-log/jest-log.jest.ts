import './index';
import logCapture from '../log-capture';
import helloWorld from '../test/hello-world';

describe('jest-log', () => {
  beforeAll(() => logCapture.start());
  afterEach(() => logCapture.reset());
  afterAll(() => logCapture.stop());

  describe('console', () => {
    test('match snapshot', () => {
      helloWorld();
      expect(logCapture).console();
    });

    test('match string', () => {
      helloWorld();
      expect(logCapture).console('INFO:  Hello World!');
    });
  });

  describe('console.info', () => {
    test('match snapshot', () => {
      helloWorld();
      expect(logCapture).console_info();
    });

    test('match string', () => {
      helloWorld();
      expect(logCapture).console_info('Hello World!');
    });

    test('match multiple lines', () => {
      helloWorld();
      console.debug('debug');
      console.error('expected error');
      expect(logCapture).console_info(
        `Hello World!
expected error`
      );
    });
  });
});

// export {}
