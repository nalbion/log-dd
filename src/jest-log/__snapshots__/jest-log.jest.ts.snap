// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-log console match snapshot: all 1`] = `"INFO:  Hello World!"`;

exports[`jest-log console.info match snapshot: info 1`] = `"Hello World!"`;
